File: Python/So_ban_be.py
def So_ban_be(a,b):
    tong1 = int();
    tong2 = int();
    for i in range(1,a,+1):
        if(a%i==0):
            tong1 = tong1 + i;
    for j in range(1,b,+1):
        if(b%j==0):
            tong2 = tong2 + j;

    """ Sau khi tính xong, ép kiểu từ số nguyên sang giá trị chuỗi"""       
    tong1 = str(tong1); tong2 = str(tong2)
    a = str(a); b = str(b)
    sbb1 = "{0}".format(tong1) # Format: Định dạng lại số, tránh 10.000000000000001
    sbb2 = "{0}".format(tong2)
    
    if(sbb1 == b):
        if(sbb2 == a):
            print("YES");
        else:
            print("NO");
    else:
        print("NO")

#main
a,b = map(int, input().split());
So_ban_be(a,b);

File: C_C++/So_ban_be.cpp
#include <iostream>
using namespace std;

void So_ban_be(int a, int b)
{
	int i,j;
	unsigned long long tong1 = 0, tong2 = 0;
	for(i=1;i<a;i++)
		if(a % i == 0)
			tong1 = tong1 + i;
	for(j=1;j<b;j++)
		if(b % j == 0)
			tong2 = tong2 + j;
	if(tong1 == b && tong2 == a)
		cout << "YES";
	else
		cout << "NO";
}
int main()
{
	int a,b;
	cin >> a >> b;
	So_ban_be(a,b);
}